src/generator/pagers.ts

```diff
@@ -47,34 +47,39 @@ export async function generatePagers(session: Session<CodeModel>): Promise<strin
     }
     text += '}\n\n';
     // pager methods
-    text += '// Err returns the last error encountered while paging.\n';
-    text += `func (p *${pager.name}) Err() error {\n\treturn p.err\n}\n\n`;
-    text += '// NextPage returns true if the pager advanced to the next page.\n'
-    text += '// Returns false if there are no more pages or an error occurred.\n';
-    text += `func (p *${pager.name}) NextPage(ctx context.Context) bool {\n`;
+    const nextLinkField = `${getResultFieldName(pager.op)}.${pager.op.language.go!.paging.nextLinkName}`;
+    text += '// More returns true if there are more pages to retrieve.\n';
+    text += `func (p *${pager.name}) More() bool {\n`;
+    text += '\tif !reflect.ValueOf(p.current).IsZero() {\n';
+    text += `\t\tif p.current.${nextLinkField} == nil || len(*p.current.${nextLinkField}) == 0 {\n`;
+    text += '\t\t\treturn false\n\t\t}\n';
+    text += '\t}\n\treturn true\n';
+    text += '}\n\n';
+
+    text += '// NextPage advances the pager to the next page.\n'
+    text += `func (p *${pager.name}) NextPage(ctx context.Context) (${respEnv}, error) {\n`;
     if (isLROOperation(pager.op)) {
       text += '\tif !p.second {\n';
       text += '\t\tp.second = true\n';
-      text += '\t\treturn true\n';
+      text += '\t\treturn p.current, nil\n';
       text += '\t} else ';
     } else {
       // note the trailing tab for the next line
       text += '\tvar req *policy.Request\n\tvar err error\n\t';
     }
     text += 'if !reflect.ValueOf(p.current).IsZero() {\n';
-    const nextLinkField = `${getResultFieldName(pager.op)}.${pager.op.language.go!.paging.nextLinkName}`;
-    text += `\t\tif p.current.${nextLinkField} == nil || len(*p.current.${nextLinkField}) == 0 {\n`;
-    text += '\t\t\treturn false\n\t\t}\n';
+    text += `\t\tif !p.More() {\n`;
+    text += `\t\t\treturn ${respEnv}{}, errors.New("no more pages")\n\t\t}\n`;
     if (isLROOperation(pager.op)) {
       text += `\t}\n\treq, err := runtime.NewRequest(ctx, http.MethodGet, *p.current.${nextLinkField})\n`;
     } else {
       text += '\t\treq, err = p.advancer(ctx, p.current)\n';
       text += '\t} else {\n';
       text += '\t\treq, err = p.requester(ctx)\n\t}\n';
     }
-    text += '\tif err != nil {\n\t\tp.err = err\n\t\treturn false\n\t}\n';
+    text += `\tif err != nil {\n\n\t\treturn ${respEnv}{}, err\n\t}\n`;
     text += `\tresp, err := p.client.pl.Do(req)\n`;
-    text += '\tif err != nil {\n\t\tp.err = err\n\t\treturn false\n\t}\n';
+    text += `\tif err != nil {\n\n\t\treturn ${respEnv}{}, err\n\t}\n`;
```

Comment:
```suggestion
    text += `\tif err != nil {\n\t\treturn ${respEnv}{}, err\n\t}\n`;
```

Reply:
Good eye, fixed.  At some point it might be nice to introduce helpers to generate things like `if err != nil` etc.